src/screens/TopicMemberScreen/index.js

```diff
@@ -54,9 +62,16 @@ const TopicMemberScreen = () => {
 
   const opacityAnimationHeader = React.useRef(new Animated.Value(1)).current;
 
+  const coverPath = topicDetail?.cover_path || null;
+  const topPosition = Platform.OS === 'ios' ? top : 0;
+
   const animatedHeight = React.useRef(
     new Animated.Value(
-      dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT + dimen.size.TOPIC_FEED_HEADER_HEIGHT
+      (coverPath
+        ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+        : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT) +
+        dimen.size.TOPIC_FEED_HEADER_HEIGHT +
+        topPosition
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for setting the `animatedHeight` seems to be repeated in multiple places (here and in the `showAnimationHeader` function). To avoid code duplication, consider creating a helper function that calculates this value based on `coverPath` and `topPosition`.
```diff
+  const calculateAnimatedHeight = () => {
+    return (coverPath ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT) + dimen.size.TOPIC_FEED_HEADER_HEIGHT + topPosition;
+  }
-  const animatedHeight = React.useRef(
-    new Animated.Value(
-      (coverPath
-        ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
-        : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT) +
-        dimen.size.TOPIC_FEED_HEADER_HEIGHT +
-        topPosition
-    )
-  ).current;
+  const animatedHeight = React.useRef(new Animated.Value(calculateAnimatedHeight())).current;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->